src/app/ryderserial/ryder-serial/responses.ts

```diff
@@ -0,0 +1,37 @@
+// Ryder device and bridge response constants.
+
+// Device responses
+
+export const RESPONSE_OK = 1; // generic command ok/received
+export const RESPONSE_SEND_INPUT = 2; // command received, send input
+export const RESPONSE_REJECTED = 3; // user input rejected
+export const RESPONSE_OUTPUT = 4; // sending output
+export const RESPONSE_OUTPUT_END = 5; // end of output
+export const RESPONSE_ESC_SEQUENCE = 6; // output esc sequence
+export const RESPONSE_WAIT_USER_CONFIRM = 10; // user has to confirm action
+export const RESPONSE_LOCKED = 11; // device is locked, send PIN
+
+// error responses
+export const response_errors: Record<number, string> = {
+  255: 'RESPONSE_ERROR_UNKNOWN_COMMAND',
+  254: 'RESPONSE_ERROR_NOT_INITIALIZED',
+  253: 'RESPONSE_ERROR_MEMORY_ERROR',
+  252: 'RESPONSE_ERROR_APP_DOMAIN_TOO_LONG',
+  251: 'RESPONSE_ERROR_APP_DOMAIN_INVALID',
+  250: 'RESPONSE_ERROR_MNEMONIC_TOO_LONG',
+  249: 'RESPONSE_ERROR_MNEMONIC_INVALID',
+  248: 'RESPONSE_ERROR_GENERATE_MNEMONIC',
+  247: 'RESPONSE_ERROR_INPUT_TIMEOUT',
+  246: 'RESPONSE_ERROR_NOT_IMPLEMENTED',
+  245: 'RESPONSE_ERROR_INPUT_TOO_LONG',
+  // 244
+  243: 'RESPONSE_ERROR_DEPRECATED',
+};
+
+// Bridge responses
+
+export const BRIDGE_RESPONSE_WAIT_IN_QUEUE: string = 'RESPONSE_WAIT_IN_QUEUE';
+export const BRIDGE_RESPONSE_BEING_SERVED: string = 'RESPONSE_BEING_SERVED';
+export const BRIDGE_RESPONSE_SHUTDOWN: string = 'RESPONSE_BRIDGE_SHUTDOWN';
+export const BRIDGE_RESPONSE_DEVICE_NOT_CONNECTED: string = 'RESPONSE_DEVICE_NOT_CONNECTED';
+export const BRIDGE_RESPONSE_DEVICE_DISCONNECTED: string = 'RESPONSE_DEVICE_DISCONNECTED';
```

Comment:
I think we should define these in the protocol doc and turn them into error codes like above.

Reply:
I was initially planning on doing that, but I ran into an issue. The bridge responses may be received at any time, even while other data is being received. So if `RyderSerial` is in the middle of reading a multi-byte response, how should a bridge response byte be distinguished from normal data? It would be possible to adjust the protocol to require escaping these bytes, but I feel that the bridge should not influence the protocol because the two are independent. Also, as we are using WebSockets here, it is simple to just use different data types to distinguish them anyways (strings in this case).

Reply:
The protocol should be adapted so that handling interrupts are specified. Interrupts will happen even more often with NFC

Reply:
The above problem @pengowen123 mentions is currently not captured in the protocol spec. From memory, when this happened we used to just send a `RESPONSE_OUTPUT_END`, which would then result in an error state of incomplete data on the receiver and then handled (hopefully).

We need a `RESPONSE_ERROR_DISCONNECT` and variants so that apps can deal with it.

@friedger right, we will have to spec protocol version 0.0.6 to deal with those issues. I think we can start early considering the above.

Maybe we can group errors, and make errors responses two bytes instead of one. So for example, an error starting with `0xFE` indicates a disconnect error and the next byte the kind.